src/vignette/http/middleware.clj

```diff
@@ -41,6 +42,19 @@
 (declare add-cache-control-header
          hours-to-seconds)
 
+(defn log-image-request
+  [handler]
+  (fn [request]
+    (let [response (handler request)]
+      (log/warn "Image request" {:status (get response :status 0)
+                             :uri (:uri request)
+                             :format (get (:query-params request) "format")
+                             :accept (get (:headers request) "accept")
+                             :agent (get (:headers request) "user-agent")
+                             :size (get (:headers response) "Content-Length" "0")
```

Comment:
I don't think `get` is necessary here. A map is also a function in Clojure. Unless you want to be a bit more verbose, I'd remove it.